src/main/java/org/folio/processing/mapping/defaultmapper/processor/Processor.java

```diff
@@ -974,37 +985,67 @@ private List<String> retrieveTargetsFromControlSubfield(DataField dataField) {
     return targets;
   }
 
+  private void buildAndFillSimpleJsonObject(Object entity, String[] embeddedFields, String value) {
+    Object currentObject = entity;
+    try {
+      currentObject = getOrCreateNestedObject(embeddedFields, currentObject);
+      setFieldValueFromPath(embeddedFields, value, currentObject);
+    } catch (Exception e) {
+      LOGGER.warn("buildSimpleJsonObject:: Error in building simple JsonObject in the mapping process: ", e);
+    }
+  }
+
+  private void setFieldValueFromPath(String[] embeddedFields, String value, Object currentObject) throws NoSuchFieldException, IllegalAccessException {
+    Field targetField = currentObject.getClass().getDeclaredField(embeddedFields[embeddedFields.length - 1]);
+    targetField.setAccessible(true);
+    targetField.set(currentObject, value);
+  }
+
+  private static Object getOrCreateNestedObject(String[] embeddedFields, Object currentObject) throws NoSuchFieldException, IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchMethodException {
+    for (int i = 0; i < embeddedFields.length - 1; i++) {
+      Field field = currentObject.getClass().getDeclaredField(embeddedFields[i]);
+      field.setAccessible(true);
+      Object nextObject = field.get(currentObject);
+      if (nextObject == null) {
+        nextObject = field.getType().getDeclaredConstructor().newInstance();
+        field.set(currentObject, nextObject);
+      }
+      currentObject = nextObject;
```

Comment:
propose not to perform this assignment to avoid complicating the logic and to return the nextObject variable. Because we are currently assigning a Dates object to the method parameter that stores Instance object.

Reply:
@RuslanLavrov  I changed this method. Feel free to look and resolve this comment.

Reply:
thank you!